darts/dataprocessing/transformers/window_transformer.py

```diff
@@ -0,0 +1,258 @@
+                )
+                or (
+                    isinstance(window_transformations, list)
+                    and len(window_transformations) > 0
+                ),
+                "`window_transformations` must be a non-empty dictionary or a non-empty list of dictionaries. ",
+            )
+
+            if isinstance(window_transformations, dict):
+                window_transformations = [
+                    window_transformations
+                ]  # if only one dictionary, make it a list
+
+            for idx, transformation in enumerate(window_transformations):
+                raise_if_not(
+                    isinstance(transformation, dict),
+                    f"`window_transformations` must contain dictionaries. Element at index {idx} is not a dictionary.",
+                )
+
+                if (
+                    "series_id" in transformation
+                    and transformation["series_id"] is None
+                ):
+                    window_transformations[idx].pop("series_id")
+
+                if (
+                    "series_id" in transformation
+                    and transformation["series_id"] is not None
+                ):
+                    raise_if_not(
+                        (
+                            isinstance(transformation["series_id"], int)
+                            and transformation["series_id"] >= 0
+                        )
+                        or (
+                            isinstance(transformation["series_id"], list)
+                            and all(
+                                isinstance(x, int) and x >= 0
+                                for x in transformation["series_id"]
+                            )
+                        ),
+                        f"`window_transformation` at index {idx} must contain a positive integer >= 0 for the "
+                        f"'series_id', or a non-empty list containing positive integers >= 0. ",
+                    )
+                    if isinstance(transformation["series_id"], int):
+                        window_transformations[idx]["series_id"] = [
+                            transformation["series_id"]
+                        ]  # make list
+
+                if (
+                    "components" in transformation
+                    and transformation["components"] is None
+                ):
+                    window_transformations[idx].pop("components")
+
+        self.window_transformations = window_transformations
+
+    def _transform_iterator(
+        self, series: Union[TimeSeries, Sequence[TimeSeries]]
+    ) -> Iterator[Tuple[TimeSeries, dict]]:
+
+        series = series2seq(series)
+
+        # run through the transformations
+        series_subset = []
+        for idx, transformation in enumerate(self.window_transformations):
+
+            if "series_id" not in transformation and "components" not in transformation:
```

Comment:
I think you could probably simplify all these cases by going along those lines (approximatively):
```python

all_series_ids = list(range(len(series)))

for idx, transformation in enumerate(self.window_transformations):
  series_ids = transformation["series_id"] if "series_id" in transformation else all_series_ids
  for s in series:
    if "components" in transformation:
      yield [(s[transformation["components"]], transformation)]
    else:
      yield [(s, transformation)]
```
(note also using `yield` for the iterator, if that's possible).